src/pages/signin/SAMLSignInPage/index.native.tsx

```diff
@@ -42,12 +75,26 @@ function SAMLSignInPage() {
             if (searchParams.has('error')) {
                 Session.clearSignInData();
                 Session.setAccountError(searchParams.get('error') ?? '');
-                Navigation.navigate(ROUTES.HOME);
+
+                Navigation.isNavigationReady().then(() => {
+                    // We must call goBack() to remove the /transition route from history
+                    Navigation.goBack();
+                    Navigation.navigate(ROUTES.HOME);
+                });
             }
         },
         [credentials?.login, shouldShowNavigation, account?.isLoading],
     );
 
+    function handleError(errorMessage: string, cleanSignInData = false) {
```

Comment:
This is a new function, can we make all params mandatory? It's clearer that way imo